### ISSUE TITLE
A Deconstruction Declaration statement should not be allowed as an embedded statement

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#13338]
```
if (...)
    var (d1, d2) = ...
```

Local declaration statement is disallowed as an embedded statement, but Deconstruction Declaration is allowed. This feels inconsistent.

**Comment by DavidArno** [comment]
It seems to me that it is the "Local declaration statement is disallowed as an embedded statement" position that is inconsistent.

If I do:
````cs
if (true)
{
    var x = "";
}
````
I get a `The variable 'x' is assigned but its value is never used` warning. But if I do:
````cs
if (true)
    var x = "";
````
I get a `Embedded statement cannot be a declaration or labeled statement` error.

**Comment by HaloFour** [comment]
@DavidArno

In your first example the block itself is the embedded statement.  The spec puts no limitation on what statements may be used within a block that happens to be an embedded statement, even if there is only a single variable declaration statement as you've demonstrated.

**Comment by gafter** [comment]
Given the grammar @MadsTorgersen gave us for deconstruction declarations, it is indeed classified as a kind of statement that isn't a valid embedded statement. This is therefore a compiler bug, not a language design question.

![image](https://cloud.githubusercontent.com/assets/3804346/17937453/fe53e856-69d6-11e6-9196-d1103bbf03f9.png)

**Comment by jcouv** [comment]
Fixed by #13379
